source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/factory/factory_env.py

```diff
@@ -0,0 +1,890 @@
+        )
+        self.step_sim_no_action()
+
+        self.randomize_initial_state(env_ids)
+
+    def _get_target_gear_base_offset(self):
+        """Get offset of target gear from the gear base asset."""
+        target_gear = self.cfg_task.target_gear
+        if target_gear == "gear_large":
+            gear_base_offset = self.cfg_task.fixed_asset_cfg.large_gear_base_offset
+        elif target_gear == "gear_medium":
+            gear_base_offset = self.cfg_task.fixed_asset_cfg.medium_gear_base_offset
+        elif target_gear == "gear_small":
+            gear_base_offset = self.cfg_task.fixed_asset_cfg.small_gear_base_offset
+        else:
+            raise ValueError(f"{target_gear} not valid in this context!")
+        return gear_base_offset
+
+    def _set_assets_to_default_pose(self, env_ids):
+        """Move assets to default pose before randomization."""
+        held_state = self._held_asset.data.default_root_state.clone()[env_ids]
+        held_state[:, 0:3] += self.scene.env_origins[env_ids]
+        held_state[:, 7:] = 0.0
+        self._held_asset.write_root_state_to_sim(held_state, env_ids=env_ids)
+        self._held_asset.reset()
+
+        fixed_state = self._fixed_asset.data.default_root_state.clone()[env_ids]
+        fixed_state[:, 0:3] += self.scene.env_origins[env_ids]
+        fixed_state[:, 7:] = 0.0
+        self._fixed_asset.write_root_state_to_sim(fixed_state, env_ids=env_ids)
+        self._fixed_asset.reset()
+
+    def set_pos_inverse_kinematics(self, env_ids):
+        """Set robot joint position using DLS IK."""
+        ik_time = 0.0
+        while ik_time < 0.25:
+            # Compute error to target.
+            pos_error, axis_angle_error = fc.get_pose_error(
+                fingertip_midpoint_pos=self.fingertip_midpoint_pos[env_ids],
+                fingertip_midpoint_quat=self.fingertip_midpoint_quat[env_ids],
+                ctrl_target_fingertip_midpoint_pos=self.ctrl_target_fingertip_midpoint_pos[env_ids],
+                ctrl_target_fingertip_midpoint_quat=self.ctrl_target_fingertip_midpoint_quat[env_ids],
+                jacobian_type="geometric",
+                rot_error_type="axis_angle",
+            )
+
+            delta_hand_pose = torch.cat((pos_error, axis_angle_error), dim=-1)
+
+            # Solve DLS problem.
+            delta_dof_pos = fc._get_delta_dof_pos(
+                delta_pose=delta_hand_pose,
+                ik_method="dls",
+                jacobian=self.fingertip_midpoint_jacobian[env_ids],
+                device=self.device,
+            )
+            self.joint_pos[env_ids, 0:7] += delta_dof_pos[:, 0:7]
+            self.joint_vel[env_ids, :] = torch.zeros_like(self.joint_pos[env_ids,])
+
+            self.ctrl_target_joint_pos[env_ids, 0:7] = self.joint_pos[env_ids, 0:7]
+            # Update dof state.
+            self._robot.write_joint_state_to_sim(self.joint_pos, self.joint_vel)
+            self._robot.reset()
```

Comment:
what is the purpose of the reset call here?

Reply:
Good catch, this is not needed. I will remove it.